lightpath/ui/gui.py

```diff
@@ -196,7 +202,7 @@ def change_path_display(self, value=None):
         """
         Change the display devices based on the state of the control buttons
         """
-        with self._lock:
+        with self._lock, self.open_splash(f'{self.selected_beamline()} path'):
```

Comment:
I had to double-check the guarantees Python gives for multiple `with` expressions:
<img width="663" alt="image" src="https://user-images.githubusercontent.com/5139267/192594868-a5fb2ffd-20d4-462d-8027-fe37c78bc609.png"> (https://docs.python.org/3/reference/compound_stmts.html)
This should be fine, then.

Thinking more about the `threading.Lock` mechanism here:
* A lock should be around a section where you don't want other threads to execute the same code at the same time
* This necessarily has to be called in the GUI thread (because it interacts with widgets)
* If somehow `change_path_display` gets called during this block, it'll cause a deadlock
* Is there some other scenario I'm not seeing?

So... I guess I'm glad the weird syntax made me think more about this code block. 
If it works fine as-is, it's probably OK to leave it...